homeassistant/helpers/template/render_info.py

```diff
@@ -0,0 +1,155 @@
+"""Template render information tracking for Home Assistant."""
+
+from __future__ import annotations
+
+import collections.abc
+from collections.abc import Callable
+from contextvars import ContextVar
+from typing import TYPE_CHECKING, cast
+
+from homeassistant.core import split_entity_id
+
+if TYPE_CHECKING:
+    from homeassistant.exceptions import TemplateError
+
+    from . import Template
+
+# Rate limiting constants
+ALL_STATES_RATE_LIMIT = 60  # seconds
+DOMAIN_STATES_RATE_LIMIT = 1  # seconds
+
+# Context variable for render information tracking
+render_info_cv: ContextVar[RenderInfo | None] = ContextVar(
+    "render_info_cv", default=None
+)
+
+
+# Filter functions for efficiency
+def _true(entity_id: str) -> bool:
+    """Return True for all entity IDs."""
+    return True
+
+
+def _false(entity_id: str) -> bool:
+    """Return False for all entity IDs."""
+    return False
```

Comment:
This is moved code so nothing to do here, but for the future: likely could be replaced with `operator.truth` and `operator.not_` which are a bit faster than the pure python function call.

Reply:
Will add that to my list to look at once I'm done with the whole refactor (the little bits to pick up at the end-list) 👍